### PR TITLE
Added Logging Guidelines

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**

### CODE REVIEW

File: docs/design/AzureSDKDesignGuidelines.md [hunk 1]
```diff
@@ -50,6 +50,32 @@ TODO: provide the full list of SDK names.
...
+
+1.2.2 :white_check_mark: **DO** emit the following log levels: Verbose (i.e. details), Informational (things happened), Warning (might be a problem or not), and Error. 
+
+1.2.3 :no_entry: **DO NOT** send sensitive information to the logs, e.g. remove account keys when logging headers.
+
+1.2.4 :white_check_mark: **DO** log a Warning message, if a request takes longer than some specified threshold. The default threshold is 3 seconds.
+
+1.2.5 :white_check_mark: **DO** log request line, response line, and headers, as Informational message.
+
+1.2.6 :white_check_mark: **DO** log an Error message, if a response comes back with a status codes between 400-599.
```
Comment (johanste) [inline docs/design/AzureSDKDesignGuidelines.md]: Another special case; [401/Unauthorized](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) during the normal authentication/challenge dance should probably be suppressed.

File: docs/design/AzureSDKDesignGuidelines.md [hunk 2]
```diff
@@ -50,6 +50,32 @@ TODO: provide the full list of SDK names.
...
+
+1.2.3 :no_entry: **DO NOT** send sensitive information to the logs, e.g. remove account keys when logging headers.
+
+1.2.4 :white_check_mark: **DO** log a Warning message, if a request takes longer than some specified threshold. The default threshold is 3 seconds.
+
+1.2.5 :white_check_mark: **DO** log request line, response line, and headers, as Informational message.
+
+1.2.6 :white_check_mark: **DO** log an Error message, if a response comes back with a status codes between 400-599. 
+
+Some services use status codes in this range in normal course of operation, e.g. implement an "exists" check by returning 404 (not fount). In such situations, the particular service might opt out from logging such status code as error.
```
Comment (johanste) [inline docs/design/AzureSDKDesignGuidelines.md]: spelling (foun*t* -> foun*d*)

File: docs/design/AzureSDKDesignGuidelines.md [hunk 3]
```diff
@@ -50,6 +50,32 @@ TODO: provide the full list of SDK names.
...
+
+Some services use status codes in this range in normal course of operation, e.g. implement an "exists" check by returning 404 (not fount). In such situations, the particular service might opt out from logging such status code as error. 
+
+1.2.7 :white_check_mark: **DO** log a Warning message, if a service call needs to be retried.
+
+1.2.7 :white_check_mark: **DO** log an Informational message, if a service call is cancelled.
+
+1.2.7 :white_check_mark: **DO** log exceptions thrown as an Error message. If the log level set to Verbose, append stack trace information to the message.
+
+1.2.7 :white_check_mark: **DO** DO log failures (exceptions and error status codes), regardless logging level, to the event log on Windows and the syslog on Linux, i.e. this cannot be turned off by the SDK user unless they replace the pipeline.
```
Comment (johanste) [inline docs/design/AzureSDKDesignGuidelines.md]: This goes against the recommendations for Python - libraries are not supposed to add any handlers (other than possibly the NullHandler). See https://docs.python.org/3.7/howto/logging.html#configuring-logging-for-a-library for more information.

Comment (bterlson) [reply docs/design/AzureSDKDesignGuidelines.md]: This is also not going to be the case for Node I think, where we should log only if given a logger (or discover the user has configured one), and should not change the pipeline. Also doesn't apply to logging failures from the browser.

**Comment by KrzysztofCwalina** [comment]
Let me merge for now. We can adjust the guidelines about syslog and eventlog logging later.

**Pull request merged**